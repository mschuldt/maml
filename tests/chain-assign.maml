arduino = Arduino(desktop=True)

@block(once)
def test():
    a = b = c = d = e = 12
    f = g = d
    f /= 6
    h = f
    print(a) # 12
    print(b) # 12
    print(c) # 12
    print(d) # 12
    print(e) # 12
    print(f) # 2
    print(g) # 12
    print(h) # 2
    die(1)

arduino.send(test)
