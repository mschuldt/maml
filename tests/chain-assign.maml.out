Initializing avm...
Ready.

12
12
12
12
12
2
12
2
bye.
